pkg/sql/plan2/query_builder.go

```diff
@@ -89,10 +105,18 @@ func (builder *QueryBuilder) remapAllColRefs(nodeId int32) (map[int64][2]int32,
 		}
 
 	case plan.Node_JOIN:
+		// collect the columns required by the current node
+		curNeedCol := newColumnCollect()
+		err := collectDepColumns(node.OnList, curNeedCol)
+		if err != nil {
+			return nil, err
+		}
+		sumNeedCol := mergeColumnCollect(curNeedCol, preNeedCol)
```

Comment:
maybe merge is not necessary。
just pass preNeedCol to collectDepColumns
like :     collectDepColumns(node.OnList, preNeedCol)